corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -984,8 +1004,10 @@ def __init__(self, domain, report_name, app_id, source_type, report_source_id, e
         self.domain = domain
 
         if self.existing_report:
+            self.registry_slug = registry_slug  # TODO: how to handle registry_slug
             self._bootstrap(self.existing_report)
         else:
+            self.registry_slug = registry_slug
```

Comment:
This is probably the best place to validate access to the data registry (or somewhere in this class)

Reply:
From some other comments above added the validations here: https://github.com/dimagi/commcare-hq/pull/30285/commits/9db442acc3d5aa2c3eb9c78676736312ee628d68
It feels a bit janky so very open to other suggestions